dns/octodns-docker/check-zone.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+'''
+Octo-DNS Reporter / exit non-zero on failure
+'''
+
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+from concurrent.futures import ThreadPoolExecutor
+from logging import getLogger
+from sys import stdout
+import sys
+from dns.exception import Timeout
+from dns.resolver import NXDOMAIN, NoAnswer, NoNameservers, Resolver, query
+import re
+
+from octodns.cmds.args import ArgumentParser
+from octodns.manager import Manager
+from octodns.zone import Zone
+
+
+class AsyncResolver(Resolver):
+    """
+    Create an DNS resolver using an async pool of workers
+    """
+
+    def __init__(self, num_workers, *args, **kwargs):
+        super(AsyncResolver, self).__init__(*args, **kwargs)
+        self.executor = ThreadPoolExecutor(max_workers=num_workers)
+
+    def query(self, *args, **kwargs):
+        return self.executor.submit(super(AsyncResolver, self).query, *args,
+                                    **kwargs)
+
+
+def main():
+    """check-zone based on octodns config file and dns zone
+    Will query all 4 DNS servers configured for the zone in GCP.
+    """
+    parser = ArgumentParser(description=__doc__.split('\n')[1])
+
+    parser.add_argument('--config-file', required=True,
+                        help='The OctoDNS configuration file to use')
+    parser.add_argument('--zone', required=True, help='zone to check')
+
+    args = parser.parse_args()
+    log = getLogger('report')
+
+    manager = Manager(args.config_file)
+    zone = Zone(args.zone, manager.configured_sub_zones(args.zone))
+    gcp = manager.providers['gcp']
+    project = gcp.gcloud_client.project
+    gcp.populate(zone)
+    #import ipdb ; ipdb.set_trace()
+
+    # Retrieve the DNS Servers directly from the GCP configuration
+    dns_servers = gcp.gcloud_zones[args.zone].name_servers
```

Comment:
At what step does this get populated?  I assume it has to reach out to GCP to find this...

Reply:
I'm pretty sure it's at [gcp.populate](https://github.com/github/octodns/blob/master/octodns/provider/googlecloud.py#L193):

```python
    manager = Manager(args.config_file)                                                                                                                                                                            
    zone = Zone(args.zone, manager.configured_sub_zones(args.zone))                                                                                                                                                
    gcp = manager.providers['gcp']                                                                                                                                                                                 
    project = gcp.gcloud_client.project                                                                                                                                                                            
    gcp.populate(zone)    
```